src/drawing-tool/components/runtime.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useRef } from "react";
+import DrawingTool from "drawing-tool";
+import 'drawing-tool/dist/drawing-tool.css';
+import { IAuthoredState, IInteractiveState } from "./app";
+import css from "./runtime.scss";
+import predefinedStampCollections from "./stamp-collections";
+import { renderHTML } from "../../shared/utilities/render-html";
+
+const kToolbarWidth = 40;
+const kToolbarHeight = 600;
+
+interface IProps {
+  authoredState: IAuthoredState;
+  interactiveState?: IInteractiveState | null;
+  setInteractiveState?: (updateFunc: (prevState: IInteractiveState | null) => IInteractiveState) => void;
+  report?: boolean;
+}
+
+interface StampCollections {
+  [collectionName: string]: string[];
+}
+
+interface DrawingToolOpts {
+  width: number;
+  height: number;
+  stamps?: StampCollections;
+}
+
+export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, setInteractiveState, report }) => {
+
+  const readOnly = !!(report || (authoredState.required && interactiveState?.submitted));
+
+  // need a wrapper as `useRef` expects (state) => void
```

Comment:
I guess you could just specify setInteractiveStateRef type as: `useRef<(updateFunc: (prevState: IInteractiveState | null) => IInteractiveState) => void>` (same as above). But it doesn't seem to matter.